src/data-model-providers/codedriven/tests/TestCodeDrivenDataModelProvider.cpp

```diff
@@ -0,0 +1,529 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <pw_unit_test/framework.h>
+
+#include <app/data-model-provider/MetadataTypes.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <app/server-cluster/ServerClusterContext.h>
+#include <app/server-cluster/ServerClusterInterface.h>
+#include <app/server-cluster/testing/TestServerClusterContext.h>
+#include <data-model-providers/codedriven/CodeDrivenDataModelProvider.h>
+#include <data-model-providers/codedriven/DynamicEndpointProvider.h>
+#include <lib/support/ReadOnlyBuffer.h>
+
+#include <memory> // For std::unique_ptr
+#include <vector> // For std::vector
+
+using chip::BitFlags;
```

Comment:
maybe at some point `using namespace chip` and `using namespace chip::app` starts being clearer.